### PR TITLE
add doc for openstack/kubernetes hybrid deploy

### PR DESCRIPTION
**Pull request by lut777**
As title

### CODE REVIEW

File: docs/OpenstackOnKubernetes.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# Openstack on Kubernetes
+
+## Target
+
+Some Kube-ovn users need to deploy both  containers and VMs(Virtual Machines) in one host node to improve resource utilization. This document describes the network part of a solution to achieve this goal.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:5]: Kube-ovn -> Kube-OVN

Comment (oilbeater) [reply docs/OpenstackOnKubernetes.md:5]: remove extra space

File: docs/OpenstackOnKubernetes.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+In this solution, Kubernetes Kube-ovn and Openstack Neutron share the same OVN platform.  Also, within a host node, both VMs and containers are connected to the same OvS(Open vSwitch). All OvS are managed by the same share OVN. Therefore, all VMs and containers are connected to the same virtual network and are scheduled by OVN.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:7]: Kube-ovn -> Kube-OVN

Comment (oilbeater) [reply docs/OpenstackOnKubernetes.md:7]: the same shared OVN

File: docs/OpenstackOnKubernetes.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+This document is based on Openstack *Victoria* and Kube-OVN *1.8.0*
+
+
+
+## Prerequisite
+
+- Openstack neutron should build networking  based on OVN.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:15]: extra space

File: docs/OpenstackOnKubernetes.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+- Subnets CIDRs in Openstack and Kubernetes connected to the same router MUST OUT be overlapped.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:16]: **MUST NOT**

File: docs/OpenstackOnKubernetes.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+
+
+## Kubernetes configuration
+
+Install Kubernetes with Kubelik-ovn in overlay mode normally.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:22]: Kube-OVN

File: docs/OpenstackOnKubernetes.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+NAME                                           STANDBY   SUBNETS
+neutron-22040ed5-0598-4f77-bffd-e7fd4db47e93   true      ["neutron-cd59e36a-37db-4c27-b709-d35379a7920f"]
+ovn-cluster                                    true      ["join","ovn-default"]
+```
+
+Obviously, the network components starting with "neutron" are from Openstack. Where "neutron-22040ed5-0598-4f77-bffd-e7fd4db47e93" is a router and "neutron-cd59e36a-37db-4c27-b709-d35379a7920f" is a subnet.
+
+##### Add subnet in Kubernetes:
+
+Now Kube-ovn supports adding new containers to the Openstack virtual network by modifying Kube-ovn CRD api. In fact, this feature comes from the support for the [vpc](https://github.com/kubeovn/kube-ovn/blob/master/docs/vpc.md).
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:102]: Kube-OVN

File: docs/OpenstackOnKubernetes.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+```
+
+There are several ways to check the correctness of the network.
+
+1. Check virtual network topology in the ovn master node.
+
+   In our example, we should check the relationship of switch provider, switch net2, and router router0 to ensure they are connected correctly.
+
+```shell
+root@ty-c3-medium-x86-02:/kube-ovn# ovn-nbctl show
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:180]: use `ko nbctl show` here

File: docs/OpenstackOnKubernetes.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+router 63d6c4d8-4aa3-41e7-8128-b8917aa6c386 (neutron-22040ed5-0598-4f77-bffd-e7fd4db47e93) (aka router0) # check router router0
+    port neutron-22040ed5-0598-4f77-bffd-e7fd4db47e93-net2
+        mac: "00:00:00:91:4D:DF"
+        networks: ["12.0.1.1/24"]
+    port lrp-a7a2f2be-c8e2-4129-b94b-f28f566e4ea6
+        mac: "fa:16:3e:1f:52:fc"
+        networks: ["192.168.1.1/24"]
+```
+
+2. Check *kube-controller-manager* to ensure there is no error.
```
Comment (oilbeater) [inline docs/OpenstackOnKubernetes.md:223]: kube-ovn-controller

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: docs/OpenstackOnKubernetes.md (hunk 1 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 2 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 3 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 4 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 5 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 6 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 7 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

File: docs/OpenstackOnKubernetes.md (hunk 8 above)
Comment (lut777) [reply docs/OpenstackOnKubernetes.md]: modified

**Review by oilbeater — APPROVED** [review]

**Pull request merged**